Src/SarifPatternMatcher/SearchSkimmer.cs

```diff
@@ -175,98 +175,150 @@ public override void Analyze(AnalyzeContext context)
 
         private void RunMatchExpression(FlexMatch binary64DecodedMatch, AnalyzeContext context, MatchExpression matchExpression)
         {
-            string searchText = binary64DecodedMatch != null
-                                   ? Decode(binary64DecodedMatch.Value)
-                                   : context.FileContents;
+            string ruleId = this.Id +
+                (!string.IsNullOrEmpty(matchExpression.SubId) ?
+                    "/" + matchExpression.SubId :
+                    string.Empty);
 
-            foreach (FlexMatch flexMatch in _engine.Matches(searchText, matchExpression.ContentsRegex))
+            FailureLevel level = matchExpression.Level != 0 ?
+                matchExpression.Level :
+                DefaultConfiguration.Level;
+
+            if (!string.IsNullOrEmpty(matchExpression.ContentsRegex))
```

Comment:
There is significant conditional statement complexity in this method, we should start to refactor into methods,

Reply:
sure! separated in two methods.

---
In reply to: [545985868](https://github.com/microsoft/sarif-pattern-matcher/pull/44#discussion_r545985868) [](ancestors = 545985868)